packages/auth/src/client/workspace.client.ts

```diff
@@ -22,7 +22,18 @@ export class WorkspaceClient {
 
     const { data } = await lastValueFrom(request)
 
-    return User.toInstance({ ...data, token: token.replace('Bearer ', '') })
+    return User.toInstance({
+      ...data,
+      token: token.replace('Bearer ', ''),
+      customRole: data.custom_role
+        ? RoleDomain.toInstance({
+            ...data.custom_role,
+            permissions: data.custom_role.permissions
+              ? PermissionDomain.toInstance({ ...data.custom_role.permissions })
+              : null,
+          })
+        : null,
```

Comment:
```suggestion
      customRole: buildCustomRole(data)
```
O que acha de isolar num método isso, só para ficar mais isolado...

Reply:
Feito! https://github.com/skore-io/nestjs-extensions/pull/175/commits/3d21b5e42db68fe83660545d176fd8abadffa931